ui/js/controllers/main.js

```diff
@@ -531,10 +531,8 @@ treeherderApp.controller('MainCtrl', [
         };
 
         const getNewReloadTriggerParams = function () {
-            return _.pick(
-                $location.search(),
-                ThResultSetStore.reloadOnChangeParameters
-            );
+            const locationSearch = $location.search();
+            return Object.assign({}, ...ThResultSetStore.reloadOnChangeParameters.map((prop) => { if (locationSearch[prop]) { return { [prop]: locationSearch[prop] }; } return false; }));
```

Comment:
Rajesh, I'm wondering if something like this would work too?
`Object.assign({}, ...ThResultSetStore.reloadOnChangeParameters.map(prop => ({[prop]: locationSearch[prop]})))`

What do you think? It may be slightly simpler?

Reply:
this will create object for all `reloadOnChangeParameters` value as key and assign `undefined` as value if suitable match value not found in `locationSearch` object.